pool/src/transaction_pool.rs

```diff
@@ -184,6 +196,50 @@ impl TransactionPool {
 		Ok(())
 	}
 
+	pub fn evict_from_txpool(&mut self, tx: &Transaction) -> Result<(), PoolError> {
+		// Order pool by fee over weight and time of the transaction
+		let mut ordered_entries = self.txpool.entries.clone();
+		ordered_entries.sort_by(|entry_a, entry_b| {
+			let fw_a = entry_a.tx.fee_weight_ratio();
+			let fw_b = entry_b.tx.fee_weight_ratio();
+			if fw_a > fw_b {
+				Ordering::Greater
+			} else if fw_a == fw_b && entry_a.tx_at < entry_b.tx_at {
+				Ordering::Greater
+			} else {
+				Ordering::Less
+			}
+		});
+
+		// Compute a list of parent transaction
+		let evictable_transactions = self.txpool.get_evictable_transactions();
```

Comment:
Maybe relying on whole buckets would allow you to evict more than childless transactions?